### PR TITLE
Jtrancos - Chall05

### PR DESCRIPTION
**Pull request by jtrancoso**
To make it simpler we only look for the instances in which the string fails the norm, which are two cases:

1. If 'ei' is not preceded by 'c'.
2. If 'ie' is preceded by 'c'.

Using `i` we go through the string and if we find 'e' or 'i' we check the next position and the previous one, if they fail the rule we return '0'.

For every other case we return '1'.

### CODE REVIEW

File: chall05/jtrancos.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include <stdio.h>
+
+int	ft_ie_except_after_c(char* str)
+{
+	int i;
+
+	i = 0;
+	while (str[i])
+	{
+		if (str[i] == 'e' && str[i + 1] == 'i' && str[i - 1] != 'c')
```
Comment (clafoutis42) [inline chall05/jtrancos.c:22]: What if `e` is the first character? You'll just go out of the string and crash  
You should try to compile with the flag `-fsanitize=address` to be more aware of such problems.

File: chall05/jtrancos.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+			return (0);
+		if (str[i] == 'i' && str[i + 1] == 'e' && str[i - 1] == 'c')
+			return (0);
+		i++;
+	}
```
Comment (clafoutis42) [inline chall05/jtrancos.c:27]: You could have used `strstr` to iterate on the character combination ;)

**Review by clafoutis42 — COMMENTED** [review]
You seem to have understood the problem, and the solution in its logic is working. But you are reading outside of the string's reserved memory, therefore you're crashing.

File: chall05/jtrancos.c (hunk 1 above)
Comment (jtrancoso) [reply chall05/jtrancos.c:22]: It could be solved by adding the condition `if (i != 0)` before both `if`, right?

File: chall05/jtrancos.c (hunk 2 above)
Comment (jtrancoso) [reply chall05/jtrancos.c:27]: Thank you for the suggestion and the evaluation :)

**Pull request merged**